filesystem/src/test/java/com/nytimes/android/external/fs3/MultiTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.nytimes.android.external.fs3
+
+import com.google.common.base.Charsets.UTF_8
+import com.google.common.collect.ImmutableMap
+import com.google.common.io.Files.createTempDir
+import com.nytimes.android.external.fs3.filesystem.FileSystem
+import com.nytimes.android.external.fs3.filesystem.FileSystemFactory
+import okio.BufferedSource
+import okio.Okio
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.io.ByteArrayInputStream
+import java.io.IOException
+import java.util.Arrays.asList
+
+class MultiTest {
+
+    @Throws(IOException::class)
+    private fun createAndPopulateTestFileSystem(): FileSystem {
+        val baseDir = createTempDir()
+        val fileSystem = FileSystemFactory.create(baseDir)
+        for (path in fileData.keys) {
```

Comment:
You can leverage destructure here and write it like:
```kotlin
for ( (path, data) in fileData) { ... }
```
and you get rid of the extra for loop. Kotlin will destructure the `MapEntry` into `path` and `data` accordingly!